test/configuration_server_test.erl

```diff
@@ -52,11 +52,38 @@ init_test() ->
     meck:expect(filelib, is_regular, fun (_Path) -> true end),
     ?assertEqual(
         {ok, {configuration, filename:absname("config.ini")}},
-        configuration_server:init([])
+        configuration_server:init([no_path])
     ),
     ?assert(meck:validate(filelib)),
     meck:expect(filelib, is_regular, fun (_Path) -> false end),
     ?assertEqual(
         {stop, {no_file, "No config file found"}},
-        configuration_server:init([])
-    ).
+        configuration_server:init([no_path])
+    ),
+    meck:unload(filelib).
+
+init_with_valid_config_flag_test() ->
+    meck:new(filelib, [unstick, passthrough]),
+    meck:expect(filelib, is_regular, fun
+        ("./tep_test.ini") -> true;
+        (Path) -> meck:passthrough([Path])
+    end),
+    ?assertEqual(
+        {ok, {configuration, "./tep_test.ini"}},
+        configuration_server:init(["./tep_test.ini"])
+    ),
+    ?assert(meck:validate(filelib)),
+    meck:unload(filelib).
+
+init_with_invalid_config_flag_test() ->
+    meck:new(filelib, [unstick, passthrough]),
+    meck:expect(filelib, is_regular, fun
+        ("./tep_test.ini") -> false;
+        (Path) -> meck:passthrough([Path])
+    end),
+    ?assertEqual(
+        {stop, {no_file, "No config file found"}},
```

Comment:
So this is a nit pick, but as a user I'd appreciate different error messages for "no file found" (because you specified none and I had to go looking for one) and "the file you specified doesn't exist".